Main.kt

```diff
@@ -0,0 +1,89 @@
+import java.lang.IllegalArgumentException
+import java.util.Scanner
+
+class Calculator(private var a: Double = 0.0, private var b: Double = 0.0) {
+    fun add() = a + b
+    fun subtract() = a - b
+    fun multiply() = a * b
+    fun divide() = if (b == 0.0) -1.0 else a / b
+}
+
+fun showMenu() {
+    println("-----Choose an option-----")
+    println("1. Add")
+    println("2. Subtract")
+    println("3. Multiply")
+    println("4. Divide")
+    println("5. Exit")
+    print("Enter your choice: ")
+}
+
+fun option() {
+    println("1. Continue to calculate with the last result")
+    println("2. Create new calculating")
+    println("3. Exit")
+    print(" Enter your option: ")
+}
+
+
+fun main() {
+    val input = Scanner(System.`in`)
+    println("Enter two number to begin calculating")
+    print("Enter a: ")
+    var a = input.next().toDouble()
+    print("Enter b: ")
+    var b = input.next().toDouble()
+    var calculator = Calculator(a, b)
+    var choice: Int? = null
+    while (true) {
+        showMenu()
+        choice = input.nextInt()
+        if (choice == 5) break
+        when (choice) {
+            1 -> {
+                println("$a + $b = ${calculator.add()}")
+                a = calculator.add()
+            }
+
+            2 -> {
+                println("$a - $b = ${calculator.subtract()}")
+                a = calculator.subtract()
+            }
+
+            3 -> {
+                println("$a * $b = ${calculator.multiply()}")
+                a = calculator.multiply()
```

Comment:
các phép gán này em nên không phải là công việc của when case này của em. Nó là của thằng showContinueActions bên dưới, vậy nên em nên chuyển xuống dưới.

Reply:
như thế thì mình vẫn phải dùng when case ngay trong hàm showContinueActions đúng k ạ? rồi trong hàm đấy mình phải truyền các biến như biến a, calculator,.. ạ?

Reply:
> Ý anh là mình gán a ngay trong hàm hay gì ạ? nma trong hàm thì mình làm công việc ấy kiểu gì ạ

```kt
when (choice) {
    1 -> result = a + b
    2 -> result = a - b
    3 -> result = a * b
    4 -> ...
}

when (choice) {
    1 -> print("$a + $b = $result")
    2 -> ...
    3 -> ...

// Người dùng chọn đi tiếp hoặc phép tính mới
if (choice == 1) {
   a = result
} else {
  // bắt đầu phép tính mới
}
```

Lí do:
1) tính tới thời điểm in ra kết quả của phép tính hiện tại, thì a vẫn nên là a, b vẫn nên là b, em không nên gán lại result cho a. Nhỡ đâu trong tương lai anh cần xem lại lịch sử các phép tính gần đây
2) gán lại a làm tăng rủi rỏ code của em hơn, nếu em muốn debug tại bước sau khi thực hiện phép toán - trước khi người dùng lựa chọn sẽ làm gì tiếp, em cần kiểm tra giá trị của a, nhưng biến a đã bị thay đổi rồi. 

Giờ chúng ta chỉ ở các bài toán cho beginner, sau này những dữ liệu của người dùng nó không chỉ còn là int với String nữa, nó sẽ là những object phức tạp: danh sách, bất đồng bộ... nên các em sẽ chưa biết được bài học của việc lười sử dụng biến immutable và val :v